src/query/api/v1/handler/search.go

```diff
@@ -46,24 +46,32 @@ const (
 
 // SearchHandler represents a handler for the search endpoint
 type SearchHandler struct {
-	store storage.Storage
+	store               storage.Storage
+	fetchOptionsBuilder FetchOptionsBuilder
 }
 
 // NewSearchHandler returns a new instance of handler
-func NewSearchHandler(storage storage.Storage) http.Handler {
-	return &SearchHandler{store: storage}
+func NewSearchHandler(
+	storage storage.Storage,
+	fetchOptionsBuilder FetchOptionsBuilder,
+) http.Handler {
+	return &SearchHandler{
+		store:               storage,
+		fetchOptionsBuilder: fetchOptionsBuilder,
+	}
 }
 
 func (h *SearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	logger := logging.WithContext(r.Context())
 
-	query, rErr := h.parseBody(r)
-	if rErr != nil {
-		logger.Error("unable to parse request", zap.Error(rErr))
+	query, parseBodyErr := h.parseBody(r)
+	opts, parseURLParamsErr := h.parseURLParams(r)
+	// NB(r): Use a loop here to avoid two err handling code paths
+	for _, rErr := range []*xhttp.ParseError{parseBodyErr, parseURLParamsErr} {
```

Comment:
Took me longer to digest this than two error handlings paths, and it possibly allocates? Would probably just do the normal thing here personally or at least do

```golang
for _, rErr := range [2]*xhttp.ParseError{parseBodyErr, parseURLParamsErr} {
...
}
```

to be sure it does not allocate

Reply:
Sure, I might actually just do a helper function like I have in `xerrors.FirstError` (only reason I didn't do that is that it expects `func FirstError(err ...error) error` but these are typed errors....